### ISSUE TITLE
Request: Training scripts & instructions for the instruction-conditioned bi-encoder (CoF)

### ISSUE THREAD

**Issue by Agenda07** [yuzhimanhua/CoF#3]
Hi, thanks a lot for releasing the code and models — amazing work!
I’m trying to **reproduce training** for the instruction-conditioned bi-encoder (semantic / topic / citation factors). I can run the embedding & evaluation scripts, but I couldn’t find the training entry points. Could you kindly clarify or share the following?

1) Training entry points
Which script/command starts training?
Example (placeholders, please correct):
`python train_cof.py \
  --dataset SIGIR \
  --factors semantic topic citation \
  --pretrained_model microsoft/BiomedNLP-BiomedBERT-base-uncased-abstract \
  --q_rep_method cls \
  --max_len 512 \
  --lr 2e-5 --batch_size 64 --epochs 20 \
  --loss info_nce --temperature 0.07 \
  --save_dir ./model/cof_sigir`
Expected folder layout for training data

2) Training data format
Exact input format(s) for semantic/topic/citation supervision:
semantic: query–doc pairs from SciRepEval (how are positives/negatives defined in your code?)
topic: MAPLE labels → how to build positive/negative pairs?
citation: triplets (p, q⁺, q⁻) — expected JSON/TSV fields? example lines?
Any pre-processing (lowercasing, dedup, max length, filtering)?
Do query IDs need to exist in *_papers_test.json as in inference?

3) Instructions (prompts)
During training, do you use the same instruction strings as in inference code (dataset-specific text for each factor)?
Max token length for instruction; do you encode the instruction once per batch or per sample?

Thank you so much for your time and for the great contribution! Any pointers (scripts, configs, or a brief recipe) would be hugely appreciated.

**Comment by yuzhimanhua** [comment]
Thank you very much for your interest in our CoF paper and for your thoughtful message.

I am very sorry, but the training code was implemented within a large internal codebase at the time, and due to my graduation, the associated data, code, and environment have already been cleaned up. Unfortunately, we no longer have an independent version of the full training pipeline.

May I ask about your specific purpose for requesting the training code? If your goal is to reproduce the results, the released checkpoint should be sufficient. If you would like to apply our model to other tasks, please feel free to treat our released model as a pre-trained scientific language model. It can take any instruction and output an instruction-specific text embedding.

I would be happy to provide additional guidance or clarifications depending on your intended use.
